File: java_src/DH3DLibrary/src/jp/spec/darkhorse2/dh3d/base/Constraint.java
/**
 * 
 */
package jp.spec.darkhorse2.dh3d.base;

/**
 * @author Norihiro
 *
 */
public class Constraint {
	public String name;
	public int bodyA;
	public int bodyB;
	public DHVector3 position;
	public DHVector3 rotate;
	public DHVector3 constraintPos1;
	public DHVector3 constraintPos2;
	public DHVector3 constraintRot1;
	public DHVector3 constraintRot2;
	public DHVector3 springPos;
	public DHVector3 springRot;
	
	public Constraint(){
		
	}
}

File: src/js/base/Constraint.js
'use strict'

/**
 * Constraint class
 * @access public
 */
export default class Constraint {
  /**
   * constructor
   * @access public
   * @constructor
   */
  constructor() {
    this.name = ''
    this.bodyA = 0
    this.bodyB = 0
    this.position = []
    this.rotate = []
    this.constraintPos1 = []
    this.constraintPos2 = []
    this.constraintRot1 = []
    this.constraintRot2 = []
    this.springPos = []
    this.springRot = []
  }
}

